src/js/components/Select/Select.js

```diff
@@ -161,10 +161,13 @@ const Select = forwardRef(
         break;
       case true:
       case undefined:
-        SelectIcon = theme.select.icons.down;
+        SelectIcon =
+          open && theme.select.icons.up
+            ? theme.select.icons.up
+            : theme.select.icons.down;
         break;
       default:
-        SelectIcon = icon;
+        SelectIcon = open && icon.up ? icon.up : icon.down || icon;
```

Comment:
I don't think that the `icon` prop knows what are 'up' and 'down', so I'm not sure if `icon.up` or `icon.down` will result anything but undefined.

Reply:
This checks for if the icon provided is an object. So, if the user gives an icon as a prop and it's an object, then it sees if they provided up or down, if they're undefined that means the user didn't provide an object and it should just behave as before.

Reply:
After discussions on this bullet, we agreed that we do not want to inflate the `icon` prop API with an object that contains up/down attributes.
Users can control the icon that is being sent to this prop and can have the logic internally on their application in case they want a separate icon for open/close of the drop.
We will roll this line up to how it was.